src/modelinstance.hpp

```diff
@@ -543,17 +543,30 @@ class ModelInstance {
 
     Status performInference(ov::InferRequest& inferRequest);
 
-    virtual Status infer(const tensorflow::serving::PredictRequest* requestProto,
-        tensorflow::serving::PredictResponse* responseProto,
-        std::unique_ptr<ModelInstanceUnloadGuard>& modelUnloadGuardPtr);
-    virtual Status infer(const ::KFSRequest* requestProto,
-        ::KFSResponse* responseProto,
+    template <typename RequestType, typename ResponseType>
+    Status infer(const RequestType* requestProto,
+        ResponseType* responseProto,
         std::unique_ptr<ModelInstanceUnloadGuard>& modelUnloadGuardPtr);
 
     ModelMetricReporter& getMetricReporter() const { return *this->reporter; }
 
     uint32_t getNumOfStreams() const;
 
     Status infer(float* data, float* output);
+
+    virtual std::unique_ptr<RequestProcessor<tensorflow::serving::PredictRequest, tensorflow::serving::PredictResponse>> getSR(const tensorflow::serving::PredictRequest*, tensorflow::serving::PredictResponse*);
```

Comment:
This is virtual because for Stateful models we require different kind of request processor.

We have 3 types declared because TFS processing wouldn't work for KFS & C-API. Ideally in the future we would sperate infer from the model instance so that it wouldn't have to know at all that TFS/KFS/C-API exist.